### PR TITLE
chore: fix merge-forward pkglint

### PR DESCRIPTION
**Pull request by njlynch**
Tried to fix this via #11672, but that didn't work. Trying invocation via `npx` instead.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by MrArnoldPalmer** [comment]
@njlynch it can't find lerna [here](https://github.com/aws/aws-cdk/blob/master/package.json#L9) right? Is pkglint guaranteed to be built by this point because the script in the root builds it as well?

**Comment by njlynch** [comment]
> Is pkglint guaranteed to be built by this point because the script in the root builds it as well?

Hmm.. so if I clean my pkglint artifacts, then run the submitted PR's version (`npx lerna...`), it breaks.

`npm run pkglint` *does* seem to first build `pkglint`, then run `lerna`. I'll update the PR to try that.
```
% npm run pkglint

> aws-cdk@0.0.0 pkglint /tmp/cdk
> lerna --scope pkglint run build && lerna run pkglint

lerna notice cli v3.22.1
lerna notice filter including "pkglint"
lerna info filter [ 'pkglint' ]
lerna info Executing command in 1 package: "yarn run build"
lerna info run Ran npm script 'build' in 'pkglint' in 20.5s:
yarn run v1.22.5
$ tsc -b && eslint . --ext=.ts && chmod +x bin/pkglint
Done in 19.77s.
lerna success run Ran npm script 'build' in 1 package in 20.5s:
lerna success - pkglint
lerna notice cli v3.22.1
lerna info Executing command in 184 packages: "yarn run pkglint"
...
```

**Review by MrArnoldPalmer — CHANGES_REQUESTED** [review]
So I think these are equivalent. My suspicion is that `npx` is needed to proceed lerna within the `pkglint` script of the root package.json. IE [this](https://github.com/aws/aws-cdk/blob/master/package.json#L9) becomes `npx lerna --scope pkglint run build && npx lerna run pkglint`.

I'm not sure whats different about the merge-forward environment that would require this, or if that script has been changed related to this. Basically I'm still not sure why this is breaking?

**Comment by njlynch** [comment]
> Basically I'm still not sure why this is breaking?

Ok, update. I've created a CodeBuild job that replicates the ForwardMerge job in my account (sans the final "create PR" step). The "aha" moments were that:
* I needed to `yarn install` first to get at our dependencies
* pkglint requires one dependency (`eslint-plugin-cdk`) to work, and needed to be built as well.

Once that was working, `pkglint` failed on `@monocdk-experiment/assert`, which is a thin shell package that mostly relies on a pre-build step to clone everything from `@aws-cdk/assert`. Disabling `pkglint` on this package was the final step. This last bit I can't test "purely", because the merge-forward script checks out `v2-main`, which wipes out my changes.

**Review by RomainMuller — APPROVED** [review]

File: scripts/merge-forward.sh [hunk 1]
```diff
@@ -8,7 +8,11 @@ git checkout -B v2-main origin/v2-main
 
 # Some package rules differ between v1 and v2, most notably which packages can be public vs private.
 # These differences are fixable via 'pkglint', so we run that and commit the delta (if any).
+yarn install --frozen-lockfile
+# Build pkglint and its dependencies (e.g., eslint-plugin-cdk), then run.
+cd tools/pkglint && ../../scripts/buildup && cd ../../
```
Comment (nija-at) [inline scripts/merge-forward.sh:13]: Can we instead update the `pkglint` script [here](https://github.com/aws/aws-cdk/blob/b16a8d3079d1d293ce127d616f82b54cb869f2e5/package.json#L9) to `lerna --scope pkglint --include-dependencies run build`?

**Review by nija-at — CHANGES_REQUESTED** [review]
> I've created a CodeBuild job that replicates the ForwardMerge job in my account (sans the final "create PR" step).

Just curious: did you try using `docker run` with the `jsii/superchain` image? Were there still gaps?

File: scripts/merge-forward.sh (hunk 1 above)
Comment (njlynch) [reply scripts/merge-forward.sh:13]: Already tried that (three iterations ago):

```
% lerna --scope pkglint --include-dependencies run build
ERR! lerna Unknown arguments: scope, include-dependencies, includeDependencies, build
```

...actually...

```
% lerna run build --scope pkglint --include-dependencies
info cli using local version of lerna
lerna notice cli v3.22.1
lerna notice filter including "pkglint"
lerna notice filter including dependencies
lerna info filter [ 'pkglint' ]
lerna info Executing command in 2 packages: "yarn run build"
...
```

That works! Odd that the first version doesn't, but the second does.

Comment (nija-at) [reply scripts/merge-forward.sh:13]: Good you found that.

**Review by nija-at — APPROVED** [review]

**Pull request merged**